theano/tensor/subtensor.py

```diff
@@ -1943,8 +1941,7 @@ def copy_of_x(self, x):
                 NPY_ARRAY_ENSURECOPY, NULL)""" % locals()
 
     def c_support_code(self):
-        from theano.gof.cutils import compile_cutils_code
-        return compile_cutils_code()
+        return inc_code()
```

Comment:
I'm not sure of this. One of the goal of the old code was to not always recompile all that code. That is why it was inside cutils. But it seem it was inside cutils and in the op! Do you agree with that interpretation? If so, keeping only 1 of the copy is good.

But currently, we recompile many ops for each input dtype, even if we include all the dtype in part of the code.

Can you add the attribute `check_input = False` to the op? Maybe it is enough to remove the duplicate thunk compiled as the support code already support many c code.